### PR TITLE
Deprecate public `window.qt_viewer` (remove in 0.5.0)

### LINKED ISSUE #3066: Make qt_viewer private
## 🧰 Task
In #1878 we discussed making `window.qt_viewer` private, so it would be `window._qt_viewer`.

We need to discuss this more before implementation as see real world usage here https://github.com/search?q=.window.qt_viewer&type=code, but it still might be worth doing as we expand the ability to do things without touching our qt files.

I'm creating this issue so we can close the existing PR, as a new one will be required anyway if we were to decide to do this

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
This PR closes #1878 and closes #3066 ...

It puts a `FutureWarning` on public access of `viewer.window.qt_viewer`, and slates it for removal in 0.5.0.

While the `QtViewer` class itself is certainly still open for use by third parties (e.g. PartSeg uses it directly maybe?)... napari's `viewer.window.qt_viewer` should be considered an implementation detail of the _napari application_, and was never designed for "public consumption".  Making this private will allow us to develop and improve the napari application with less fear of breaking patterns (i.e. you know if you're using a private attribute that it's a bit risky), and there's been general support for this (at least from @sofroniewn, @jni, and myself) for quite a while.  Specific hooks will be made available to add things like menus via plugins and npe2.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
Love this, thanks so much @tlambert03 - will be great to get this in. I'd like to merge this asap, but will wait for approval from @jni too

**Comment by sofroniewn** [comment]
Just created another merge conflict, sorry!! Will merge this one at next possible moment though. Thanks for all the updates here

File: napari/_tests/test_interactive_transforms.py
Comment (jni) [inline napari/_tests/test_interactive_transforms.py:111]: Have I mentioned how much I hate black??? 😂

**Review by jni — APPROVED** [review]

File: napari/_tests/test_interactive_transforms.py
Comment (tlambert03) [reply napari/_tests/test_interactive_transforms.py:111]: Lol, yeah that's a bad one

**Comment by sofroniewn** [comment]
fails here are unrelated

**Comment by sofroniewn** [comment]
oh shoot, i was supposed to wait to merge this until we fixed examples. sorry @jni @tlambert03 - complete memory fail. i saw @jni approve and i just went for it. i think it's ok to press on, i just saw #3765 which reminded me

**Comment by tlambert03** [comment]
No worries, will follow up

**Pull request merged**
